kotsadm/web/src/components/apps/AppSnapshotSchedule.jsx

```diff
@@ -118,17 +121,43 @@ class AppSnapshotSchedule extends Component {
     this.setState({ selectedRetentionUnit: retentionUnit });
   }
 
-  componentDidUpdate = (lastProps) => {
-    if (this.props.snapshotConfig.snapshotConfig && this.props.snapshotConfig.snapshotConfig !== lastProps.snapshotConfig.snapshotConfig) {
+  getSnapshotConfig = async () => {
+    this.setState({ loadingConfig: true, gettingConfigErrMsg: "", displayErrorModal: false });
+    await fetch(`${window.env.API_ENDPOINT}/app/${this.props.app.slug}/snapshot/config`, {
+      method: "GET",
+      headers: {
+        "Authorization": Utilities.getToken(),
+        "Content-Type": "application/json",
+      }
+    }).then(async (res) => {
+      if (!res.ok) {
+        this.setState({ loadingConfig: false, gettingConfigErrMsg: `Unable to get snapshot config: Unexpected status code: ${res.status}`, displayErrorModal: true });
+        return
+      }
+      const body = await res.json();
+      this.setState({
+        snapshotConfig: body,
+        loadingConfig: false
+      })
+    }).catch((err) => {
+      console.log(err);
+      this.setState({ loadingConfig: false, gettingConfigErrMsg: err ? err.message : "Something went wrong, please try again.", displayErrorModal: true });
+    });
+  }
+
+  componentDidUpdate = (lastProps, lastState) => {
+    if (this.state.snapshotConfig && this.state.snapshotConfig !== lastState.snapshotConfig) {
       this.setFields();
     }
   }
 
   componentDidMount = () => {
-    this.getReadableCronExpression();
-    if (this.props.snapshotConfig.snapshotConfig) {
+    if (this.state.snapshotConfig) {
       this.setFields();
+    } else {
+      this.getSnapshotConfig()
```

Comment:
writing in go made you forget a few `;` 🙂

Reply:
haha ah yeah the switching back and forth definitely has me forgetting those!